src/init.cpp

```diff
@@ -1027,6 +1031,12 @@ bool AppInitParameterInteraction()
         LogPrintf("Warning: nMinimumChainWork set below default value of %s\n", chainparams.GetConsensus().nMinimumChainWork.GetHex());
     }
 
+    {
+        const int64_t pct = gArgs.GetArg("-dandelion", DEFAULT_DANDELION_STEM_PERCENTAGE);
+        if (pct < 0 || 100 < pct) return InitError(_("-dandelion must be 0 or a percentage less than 100"));
```

Comment:
Should `-dandelion=100` generate an error message? From the error message it looks like it should, but not from the logic :-)

Reply:
100% is allowed to always extend the stem by one hop. So the interval is [0, 100].

Reply:
Then the error message should be `-dandelion must be 0 or a percentage up to 100` or something equivalent?